app/controllers/accounts_controller.rb

```diff
@@ -1,4 +1,8 @@
+# frozen_string_literal: true
+
 class AccountsController < ApplicationController
+  before_action :authenticate_user!
```

Comment:
The public shared account should be available for **all** people (including not logged in). That's why they are public.